Misc/NEWS.d/next/Library/2024-08-19-09-36-50.gh-issue-123135.w_ZNAs.rst

```diff
@@ -0,0 +1,2 @@
+Fix :term:`path-like objects <path-like object>` support in
+:func:`fnmatch.filter`.  Patch by Bénédikt Tran.
```

Comment:
I think this is a feature, so it could be phrased:

```suggestion
Add support for supplying :term:`path-like objects <path-like object>`
to the *names* argument of :func:`fnmatch.filter`. Previously,
such objects were accepted only on Windows. Patch by Bénédikt Tran.
```

Reply:
Personally I consider it as a bugfix but I changed it into a feature. The reason I considered it a bug fix in the first place is because we said that it's equivalent to `[name for name in names if match(normcase(name))]` and `normcase()` accepts path-like objects.